bpf/bpf_lxc.c

```diff
@@ -936,6 +944,12 @@ ipv6_policy(struct __ctx_buff *ctx, int ifindex, __u32 src_label, __u8 *reason,
 	if (skip_ingress_proxy)
 		verdict = 0;
 
+	if (unlikely(ret == CT_REOPENED)) {
+		send_policy_verdict_notify(ctx, src_label, tuple.dport,
+					   tuple.nexthdr, POLICY_INGRESS, 1,
+					   verdict, policy_match_type, audited);
+	}
```

Comment:
Can't we merge this with the `ret == CT_NEW` case below? Same for the ipv4 path.

Reply:
Thanks Paul for the review! The case ret == CT_NEW will do more than what needs to be done for ret == CT_REOPENED. Do you mean we merge it and then have another check inside that? i.e.
if (ret == CT_NEW || ret == CT_REOPENED) {
    send_policy_verdict_notify()
    if (ret == CT_NEW) {
        ....
   }
}

Reply:
No, you're right. Not sure how I misread this given if basically the same code in all cases :confused: 

I also considered merging the longer `CT_NEW` cases with the new `CT_REOPENED` case, but that would likely need to slightly more harder-to-read code for little gain. LGTM!

Reply:
I was wondering about this too, since IMO it's harder to tell whether the parameters are identical than it is to read two conditionals - `if (new || reopened) send_trace_notify(...); ` followed by `if (new)` for the rest of the existing "new" handling code.

My inclination would be that avoiding the "stutter" with an approach similar to what @lzang outlined in the comment above would be shorter and simpler to read without making things much more complicated.

Reply:
I was meant to reduce the number of comparison, and code duplication cannot be avoided in switch case. But I agree that the alternative might be easier to read. I am fine either way.

Reply:
yeah for switch case it needs to be duplicated anyway. I think for the regular if{} conditionals it'd be marginally tidier but I'm not so worried about it.

Reply:
I just changed the logic to if (new || reopened). please take a look and let me know if this patch is ready to merge. Thanks!